slang-kotlin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -134,6 +140,9 @@ private Tree createElement(@Nullable PsiElement element) {
         // FIXME are we sure we want body of function as block tree ?
         bodyTree = new BlockTreeImpl(bodyTree.metaData(), Collections.singletonList(bodyTree));
       }
+      if (bodyTree != null && bodyTree.children().isEmpty()) {
+        bodyTree = null;
```

Comment:
I think this if can be integrated somewhat with the previous one.
Also, a comment on why we do it would be nice too. I know this is related to empty lambda, but it might not be obvious to someone who did not investigate the issue.